src/pages/postgraphile/usage-schema.md

```diff
@@ -238,3 +238,288 @@ look at the `postgraphile-core` and `graphile-build-pg` modules.
+- optional: `@graphql-codegen/typescript-urql` - this produces typescript
+  queries from any custom `*.graphql` query/mutation files that we define.
+
+Steps to get the backend typing support:
+
+1. Start the development like described in the above section. Follow all the
+   steps to create your first GraphQL API with all needed schemas, tables,
+   roles, etc.
+2. Configure in the PostGraphile setup to export the GraphQL schema:  
+   `exportGqlSchemaPath: './src/generated/schema.graphql'`
+3. Start the project to let PostGraphile create the initial `schema.graphql`
+   file. You can exclude this file from source control if you want to. But it
+   might be nice to see differences in the schema during check-ins.
+4. Import the mentioned `npm` packages. You can find more plugins on their
+   website in the [Plugins](https://graphql-code-generator.com/docs/plugins/)
+   section.
+5. Create a file `codegen.yml` in the root of your workspace.
+   ```yaml
+   overwrite: true
+   schema: "./src/generated/schema.graphql"
+   generates:
+     # Creates the TypeScript types from the schema and any .graphql file
+     src/generated/types.ts:
+       documents: "src/**/*.graphql"
+       plugins:
+         - typescript
+         - typescript-operations
+         - typescript-urql
+       config:
+         withHOC: false
+         withComponent: false
+         withMutationFn: false
+   config:
+     scalars:
+       DateTime: "string"
+       JSON: "{ [key: string]: any }"
+   ```
+6. Run `graphql-codegen --config codegen.yml` to generate the types.
+7. The generated types can now be used in your custom business logic code.
+
+### Example
+
+We have a movie table that we want to query from our backend system.
+
+We can write a small GraphQL query file similar to this. It could be stored in
+`./src/graphql/getMovies.graphql`.
+
+```graphql
+query GetMovies($top: Int!) {
+  movies(first: $top) {
+    nodes {
+      id
+      title
+    }
+  }
+}
+```
+
+Save the file and run the code generation task.
+
+Alternatively we can create some inline query directly in code like this:
+
+```js
+import { makeExtendSchemaPlugin, gql } from "graphile-utils";
+
+// inside some function:
+
+const GetMoviesDocument = gql`
+  query Query {
+    __typename
+    movies(first: 3) {
+      nodes {
+        id
+        title
+      }
+    }
+  }
+`;
+```
+
+The query can then be used in your code via the generated type or inline:
+
+```js
+
+  // This would be generated from the above .graphql file by the code generator
+  // alternatively define it inline via gql` like shown above
+  const doc = GetMoviesDocument;
+  const variables: GetMoviesQueryVariables = {
+    top: 3,
+  };
+
+  // Execute the query from your backend code
+  // The "Query" type is your the root Query type. It contains all the query
+  // endpoints from the GraphQL API with type support
+  const queryResult = await graphql.execute<Query>(
```

Comment:
I don't think the type is correct here; I think this is used to describe the shape of the result which should be the TypeScript type that graphql-code-generator generated from the `GetMovies` operation via the `@graphql-codegen/typescript-operations` plugin:

```suggestion
  const queryResult = await graphql.execute<GetMovies>(
```

Reply:
Yes of course - thanks for pointing out. Should be `GetMoviesQuery`.